python-package/xgboost/spark/core.py

```diff
@@ -281,19 +283,40 @@ def _validate_params(self):
                 .get("spark.task.resource.gpu.amount")
             )
 
-            if not gpu_per_task or int(gpu_per_task) < 1:
-                raise RuntimeError(
-                    "The spark cluster does not have the necessary GPU"
-                    + "configuration for the spark task. Therefore, we cannot"
-                    + "run xgboost training using GPU."
-                )
+            is_local = _is_local(
+                _get_spark_session()
+                .sparkContext
+            )
 
-            if int(gpu_per_task) > 1:
-                get_logger(self.__class__.__name__).warning(
-                    "You configured %s GPU cores for each spark task, but in "
-                    "XGBoost training, every Spark task will only use one GPU core.",
-                    gpu_per_task
-                )
+            if is_local:
+                # checking spark local mode.
+                if gpu_per_task:
+                    raise RuntimeError(
+                        "The spark cluster does not support gpu configuration for local mode. "
+                        + "Please delete spark.executor.resource.gpu.amount and "
+                        + "spark.task.resource.gpu.amount"
+                    )
+
+                if self.getOrDefault(self.num_workers) > 1:
+                    raise ValueError(
+                        "Training XGBoost on the spark local mode only supports num_workers = 1, "
+                        + "and only primary GPU device will be used."
+                    )
```

Comment:
I think we can still support multiple GPU workers in spark local mode.
We can get partition id from spark TaskContext, and use the partition id as the gpu_id for the corresponding spark task. WDYT ?

Reply:
Thx. Hmm, there will be only 1 process for spark local mode, If we support the scenario of num_workers > 1, then all the GPU training tasks will run on the same process, which seems not to be supported due to the NCCL issue. @trivialfis Could you confirm that?

Reply:
It should work but let's not invite trouble.

Reply:
Yeah, I agree. The GPU supporting for the local mode is mostly used in local debugging.

Reply:
@wbo4958 
A correction:
> local mode pyspark all the GPU training tasks will run on the same process

This is not true.
For pyspark, in barrier mode pyspark task, each task will be run in an individual python process.

Reply:
Oh right, my bad. I previously thought it was the JVM side. Ok, let me add it support.